src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -759,24 +759,30 @@ private boolean isSameParameter(CtMethod<?> method, CtTypeReference<?> ctParamet
 				return false;
 			}
 		}
+		if (ctParameterType.equals(factory.Type().OBJECT) && (expectedType instanceof CtTypeParameterReference)) {
+			CtTypeParameterReference typeParameterReference = (CtTypeParameterReference) expectedType;
+			if (!typeParameterReference.isUpper() || (typeParameterReference.isUpper() && typeParameterReference.getBoundingType().equals(ctParameterType))) {
+				return true;
+			}
+		}
 		if (expectedType instanceof CtTypeParameterReference && ctParameterType instanceof CtTypeParameterReference) {
 			// Check if Object or extended.
 			if (!ctParameterType.equals(expectedType)) {
 				return false;
 			}
 		} else if (expectedType instanceof CtTypeParameterReference) {
-			if (!ctParameterType.isSubtypeOf(factory.Type().createReference(expectedType.getActualClass()))) {
+			if (!ctParameterType.isSubtypeOf(factory.Type().createTypeParameterReference(expectedType.getQualifiedName()))) {
```

Comment:
I do not understand the purpose of both - old and new code. Could anybody explain it?